### ISSUE TITLE
Update remaining DockerHub-hosted repositories to use ECR

### ISSUE THREAD

**Issue by 007** [aws/eks-charts#356]
tl;dr: Please update `amazon/foo` registry values in these charts to use Amazon-hosted ECR versions of the same.

Per the [Coming Soon blogpost](https://aws.amazon.com/blogs/containers/advice-for-customers-dealing-with-docker-hub-rate-limits-and-a-coming-soon-announcement/) for the AWS public registry:

> All EKS add-on software that is included with EKS clusters is hosted on ECR today and will not be subject to Docker Hub rate limits. EKS-built and maintained projects, such as the AWS Load Balancer Controller include helm charts that reference images on ECR as well. However, manifests for these projects available on GitHub include references to images that are published to Docker Hub. You can check whether you are using a manifest with a Docker Hub image using the steps described in this blog and update to use a chart that references an ECR hosted image. These charts are available in the [EKS Charts](https://github.com/aws/eks-charts) GitHub repository.

We went to update our `aws-node-termination-handler` to use this chart after getting DockerHub pull failures, but it doesn't look like it's going to make a difference:

https://github.com/aws/eks-charts/blob/d721c97c339c8b1060ac8b76a247ce67a470e90c/stable/aws-node-termination-handler/values.yaml#L6

There are several other places that would be subject to the same throttling:
```
stable/appmesh-controller/values.yaml:45:    repository: amazon/aws-xray-daemon
stable/aws-cloudwatch-metrics/values.yaml:2:  repository: amazon/cloudwatch-agent
stable/aws-for-fluent-bit/values.yaml:6:  repository: amazon/aws-for-fluent-bit
stable/aws-node-termination-handler/values.yaml:6:  repository: amazon/aws-node-termination-handler
```
And several public images that might need to be replicated as well, either as part of the deployment or as part of CI:
```
stable/appmesh-controller/ci/values.yaml:7:  repository: fawadkhaliq/appmesh-controller
stable/appmesh-gateway/ci/values.yaml:5:  repository: envoyproxy/envoy
stable/appmesh-grafana/values.yaml:8:  repository: grafana/grafana
stable/appmesh-inject/ci/values.yaml:5:  repository: stefanprodan/aws-app-mesh-inject
stable/appmesh-jaeger/values.yaml:8:  repository: jaegertracing/all-in-one
stable/appmesh-prometheus/values.yaml:11:  repository: prom/prometheus
stable/aws-load-balancer-controller/ci/values.yaml:5:  repository: kishorj/aws-load-balancer-controller
```

Please update the remaining `amazon/*` images (at least) to use an AWS ECR repository value instead of relying on the current DockerHub repo.

**Comment by bwagner5** [comment]
aws-node-termination-handler will be moving to ECR images soon.

**Comment by fawadkhaliq** [comment]
AWS ECR public gallery is out: https://gallery.ecr.aws/

We'll be moving the images to ECR gallery

**Comment by bwagner5** [comment]
NTH Public ECR: https://gallery.ecr.aws/r6b0f9a1/aws-node-termination-handler

Chart will be updated soon to use it by default. Images will still be pushed to Dockerhub in addition to ECR for now.

**Comment by mamoit** [comment]
The cloudwatch-agent's image tag that comes by default in the chart is not available in the public repository.
It seems to be too old.
```
public.ecr.aws/cloudwatch-agent/cloudwatch-agent:1.247345.36b249270
```

Should the default image tag be bumped?
And maybe even setting the default registry to be the public one to not depend on dockerhub.
